backend/collect/graphs.py

```diff
@@ -0,0 +1,44 @@
+from typing import List
+from rdflib import Graph, RDF, IdentifiedNode
+from rdflib.term import Node
+
+from triplestore.utils import Triples
+
+def list_from_graph_collection(graph: Graph, list_node: IdentifiedNode) -> List[Node]:
+    '''
+    Extract a list of nodes from an RDF collection in a graph
+    '''
+
+    items = list(graph.objects(list_node, RDF.first))
+    rest_nodes = graph.objects(list_node, RDF.rest)
+    for rest in rest_nodes:
```

Comment:
From what I can tell, there should be only one `rest` to a list, but nothing will enforce that on a technical level. (The [specification of rest](https://www.w3.org/TR/rdf12-schema/#ch_rest) does not mention such a restriction, though.)

Anyway, *if* that happens, this solution is to concatenate the different lists, but another option would be to pick one and ignore the rest.

Now that you mention it, the recursive definition is a bit problematic since you can easily define circular lists in RDF - which would lead to infinite recursion. This can also happen if you drop all but one `rest` element.

I don't think this is a huge issue here, though, since edpop collections are created by the application itself. It would be an issue with a generic collection reader, though.

Reply:
Ideally this algorithm should use a set membership check to prevent an infinite loop. Up to you whether you want to implement that on this branch or defer it to a new issue.